lisa/tools/kvm_unit_tests.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import string
+from pathlib import Path, PurePath
+from typing import List, Type, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+
+from .git import Git
+from .make import Make
+
+
+class KvmUnitTests(Tool):
+    test_runner_cmd: str = ""
+    repo_root: PurePath
+
+    repo = "https://gitlab.com/kvm-unit-tests/kvm-unit-tests.git"
+    deps = [
+        "gcc",
+        "make",
+        "binutils",
+        "qemu-kvm",
+    ]
+
+    @property
+    def command(self) -> str:
+        return str(self.test_runner_cmd)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Make]
+
+    @property
+    def exists(self) -> bool:
+        return True if self.test_runner_cmd else False
+
+    def _install_dep(self) -> None:
+        posix_os: Posix = cast(Posix, self.node.os)
+        tool_path = self.get_tool_path()
+        git = self.node.tools[Git]
+        git.clone(self.repo, tool_path)
+
+        # install dependency packages
+        for package in list(self.deps):
+            if posix_os.is_package_in_repo(package):
+                posix_os.install_packages(package)
+
+    def _install(self) -> bool:
+        self._log.info("Building kvm-unit-tests")
+        self._install_dep()
+        tool_path = self.get_tool_path()
+        make = self.node.tools[Make]
+        self.repo_root = tool_path.joinpath("kvm-unit-tests")
+
+        # run ./configure in the repo
+        configure_path = self.repo_root.joinpath("configure")
+        self.node.execute(str(configure_path), cwd=self.repo_root, expected_exit_code=0)
+
+        # run make in the repo
+        make.make("", self.repo_root)
+
+        self.test_runner_cmd = str(self.repo_root.joinpath("run_tests.sh"))
+        self._log.info("Finished building kvm-unit-tests")
+        return True
```

Comment:
What's the point of doing it that way? Here we need to return true if the installation succeeded. The `make` command succeeded so we can directly return True.

Reply:
This command is not added to PATH. The command here is a script called `run_tests.sh` in the kvm-unit-tests repo. We invoke it directly by giving a full path to it. The "installation" in this case just involves building (`make`) the code so that the unit test binaries are generated. So, installation has succeeded if `make` has succeeded.

Reply:
The current implementation may not be able to detect the tool is installed already. It means, if the environment is used second time, the tool may be installed again. That's why it needs the _check_exists. It's not a good practice to set the command in `_install`, because if it's not called, the command is not set. The command should be assigned in `_initialize`, if it's possible.

And echo Divyansh's point, the `_check_exists` should be called after installation. There may be unknown reason to fail the installation, we should know it as early as possible.

Reply:
Done